packages/frontend/src/pages/NewListing.tsx

```diff
@@ -0,0 +1,421 @@
+              <FormControl>
+                <Input
+                  {...field}
+                  type='text'
+                  value={displayPrice}
+                  className='w-[180px] text-base'
+                  onChange={(e) => {
+                    const value = parseFloat(e.target.value.replace(/[^\d\.]/g, ''));
+                    if (!isNaN(value)) {
+                      setValue('price', value);
+                      setDisplayPrice(e.target.value)
+                    }
+                  }}
+                  onBlur={async () => {
+                    await trigger('price');
+                    field.onBlur();
+                    setDisplayPrice(new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD' }).format(price));
+                  }}
+                />
+              </FormControl>
+              {errors && <FormMessage>{errors.price?.message}</FormMessage>}
+            </FormItem>
+          )
+        }}
+      />
+      <FormField
+        control={control}
+        name="description"
+        render={({ field }) => (
+          <FormItem>
+            <FormLabel className='text-base'>Description</FormLabel>
+            <FormControl>
+              <Textarea className='text-base'
+                {...field}
+                onBlur={async (e) => {
+                  await trigger('description');
+                  field.onBlur();
+                }}
+              />
+            </FormControl>
+            {errors && <FormMessage>{errors.description?.message}</FormMessage>}
+          </FormItem>
+        )}
+      />
+    </section >
+  )
+}
+
+const TrustScoreFormStep = ({ control }: StepSectionProps) => {
+  return (
+    <section className='flex flex-col space-y-4'>
+      {Object.entries(trustScoresFromData).map(([key, scoreInfo]) => (
+        <FormField
+          key={key}
+          control={control}
+          name={`revealTrustScores.${key}`}
+          render={({ field }) => (
+            <FormItem className='flex justify-between items-start space-x-6 border border-muted-foreground p-3' key={key}>
+              <div>
+                <FormLabel className="text-foreground text-lg" htmlFor={key}>
+                  {scoreInfo.title}:{' '}<span className="text-blue-700 font-extrabold">{scoreInfo.score}%</span>
+                </FormLabel>
+                <FormDescription className='text-foreground/80'>{scoreInfo.description}</FormDescription>
+              </div>
+              <FormControl>
+                <div className="flex space-x-2">
+                  <Switch
+                    id={key}
+                    checked={field.value}
+                    onCheckedChange={field.onChange}
+                    className={cn(field.value ? 'data-[state=checked]:bg-blue-700' : '')}
+                  />
+                  <p className='text-muted-foreground'>{field.value ? 'Show' : 'Hide'}</p>
+                </div>
+              </FormControl>
```

Comment:
how can we use the switch to change the stored score to an X if the user chooses to hide?

if we can send like scores = {LP : 45, LO: 89, CB: X, GV: n/a} as a string to the database, i think i can figure out how to read the scores into the other components that display them, without having to rewrite too much. also i think i will be able to get rid of that second crazy math function in the Trustlist hook 😂

Reply:
I don't think we should to remove `revealTrustScores`. We can just use it to determine the actual scores we'll send to the API in  the `publishListing` fn